stdlib/tests/collections/smt.rs

```diff
@@ -0,0 +1,162 @@
+    let advice_map = [];
+    let expected = [ZERO; WORD_SIZE];
+    assert_smt_get_opens_correctly(key, expected, initial_root, store, &advice_map);
+
+    // check included leaves for all tiers
+    for depth in [16, 32, 48, 64] {
+        // init a new storage
+        let mut store = MerkleStore::new();
+
+        // compute the index and node value
+        let index = key[3].as_int();
+        let index = index >> (64 - depth);
+        let index = NodeIndex::new(depth as u8, index).unwrap();
+        let remaining = get_smt_remaining_key(key, depth as u8);
+        let node =
+            Rpo256::merge_in_domain(&[remaining.into(), value.into()], Felt::new(depth)).into();
+
+        // insert the target node into the storage
+        let root = store.set_node(initial_root, index, node).unwrap().root;
+
+        // compute the index and sibling value
+        let sibling = key[3].as_int();
+        let sibling = sibling >> (64 - depth);
+        let sibling = sibling ^ 1;
+        let sibling = NodeIndex::new(depth as u8, sibling).unwrap();
+        let mut sibling_node = Word::from(node);
+        sibling_node[3] = sibling_node[3] + ONE;
+        let mut sibling_value = Word::from(value);
+        sibling_value[3] = sibling_value[3] + ONE;
+
+        // insert the sibling node into the storage
+        let root = store.set_node(root, sibling, sibling_node).unwrap().root;
+
+        // execute the test and expect the right node to be returned
+        let advice_map = [
+            (node.into_bytes(), value.to_vec()),
+            (sibling_node.into_bytes(), sibling_value.to_vec()),
+        ];
+        assert_smt_get_opens_correctly(key, value, root, store, &advice_map);
+    }
+}
+
+#[test]
+fn smtget_opens_correctly_for_empty_leaf() {
+    // setup the test
+    let empty = EmptySubtreeRoots::empty_hashes(64);
+    let mut seed = 1 << 50;
+
+    // test different cases where the path diverges on the target
+    for diverge in [16, 32, 48, 64] {
+        // setup a new store
+        let mut store = MerkleStore::new();
+        let root = Word::from(empty[0]);
+
+        // generate a new key/value pair
+        let key = seeded_word(&mut seed);
+        let value = seeded_word(&mut seed);
+
+        // compute a sibling index and node value
+        let mut sibling = key[3].as_int();
+        sibling ^= 1 << (64 - diverge);
+        let remaining = get_smt_remaining_key(key, diverge as u8);
+        let sibling = NodeIndex::new(diverge as u8, sibling >> (64 - diverge)).unwrap();
+        let node =
+            Rpo256::merge_in_domain(&[remaining.into(), value.into()], Felt::new(diverge)).into();
+
+        // expect ZERO as the leaf isn't included
+        let root = store.set_node(root, sibling, node).unwrap().root;
+        let advice_map = [(node.into_bytes(), value.to_vec())];
+        let expected = [ZERO; WORD_SIZE];
+        assert_smt_get_opens_correctly(key, expected, root, store.clone(), &advice_map);
+    }
```

Comment:
Similar to the last comment: I'd probably build the `MerkleStore` separately (could have the exact same structure as the one form the previous comment) and then tests key lookups where the keys differ in just one bit from what's in the store.

Reply:
Tests refactored in https://github.com/0xPolygonMiden/miden-vm/pull/808#discussion_r1162275098